client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -0,0 +1,59 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get, noop } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice } from 'state/notices/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { JETPACK_ONBOARDING_SETTINGS_SAVE } from 'state/action-types';
+
+/**
+ * Dispatches a request to save particular onboarding settings on a site
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const saveJetpackOnboardingSettings = ( { dispatch, getState }, action ) => {
+	const { settings, siteId } = action;
+	const state = getState();
```

Comment:
Usually trying to avoid fetching `state` inside actions. However, the only other way I can think would mean not storing credentials in Redux state at all but, e.g., again in the current URL (as a query string), which isn't any more elegant I guess.